common/shared.c

```diff
@@ -86,6 +86,7 @@ static int getaddr(char *node, char *service,
 {
 	int ret;
 	struct fi_info *fi;
+	struct fi_info hcopy = *hints;
```

Comment:
This doesn't properly copy fi_info.  If we need an actual copy, use fi_dupinfo.  Otherwise, just pass in hints.